pkg/ingester/instance.go

```diff
@@ -971,6 +973,7 @@ func sendBatches(ctx context.Context, i iter.EntryIterator, queryServer QuerierQ
 
 		stats.AddIngesterBatch(int64(batchSize))
 		batch.Stats = stats.Ingester()
+		batch.Warnings = metadata.Warnings()
```

Comment:
Here's where we attach warnings to the ingester batch response. From here on out, it's a matter of encoding and decoding (and encoding and decoding) in the scheduler and frontend to propagate them to the user